### ISSUE TITLE
Button improvement requests

### ISSUE THREAD

**Issue by mendeza** [ITS-HCD/nysds#1464]
# Overview

- The Button component has attribute `tabindex="0"`.
  - **Request**: this is natively provided and can be removed.
- The Button component, in its default mode as a 'button', has `role="button"`.
  - **Request**: this is redundant and can be removed. 
- The Button component renders `aria-label=<labelvalue>` and doesn't allow authors to set an arbitrary `aria-label` value. The content of `.nys-button__text` is the accessible name and recognized by ATs. Thus the `aria-label`, which overrides it, is redundant and heavy-handed. 
  - Consideration: a method exists to work around this when a `circle` button has only an icon as visible content. In the assumption that the team does not wish authors to make full-size icon buttons, this still leaves the scenario when additional context is needed for ATs.
  - **Request**: with `circle` button, keep the labeltext but make it __visually-hidden__, e.g.,  `<div class="nys-button__text sr-only">labeltext</div>`
  - **Request**: Allow arbitrary user-provided `aria-label` values to be passed down from the WC to the shadow-root. And while at it, `aria-describedby` also.

## Annotations

- **`[A]`** &rarr; failure of ARIA association
- **`[B]`** &rarr; it makes sense that the WC attribute `label` maps to the accessible text content of the rendered HTML
- **`[C]`** &rarr; `<button>` doesn't need `role="button"`
- **`[D]`** &rarr; `tabindex="0"` is already a native behavior for these elements
- **`[E]`** &rarr; `aria-label` values rendered in the shadow-root ignore authors' legitimate need to set arbitrarily

<img width="100%" alt="Annotated screenshot of buttons and their rendered code showing poor use of attributes" src="https://github.com/user-attachments/assets/7bfe98b2-346a-407a-8a1d-72dddd0c5edc" />

**Comment by mendeza** [comment]
I retract the request to remove `type="button"`; that's a good default, and users should have the ability to change it when necessary.
